src/Nethermind/Nethermind.Serialization.Json/LongConverter.cs

```diff
@@ -37,19 +38,16 @@ public LongConverter(NumberConversion conversion)
 
         public override void WriteJson(JsonWriter writer, long value, Newtonsoft.Json.JsonSerializer serializer)
         {
-            if (value == 0L)
-            {
-                writer.WriteValue("0x0");
-                return;
-            }
-            
             switch (_conversion)
             {
                 case NumberConversion.Hex:
-                    writer.WriteValue(value.ToHexString(true));
+                    writer.WriteValue(value == 0L ? "0x0" : value.ToHexString(true));
                     break;
                 case NumberConversion.Decimal:
-                    writer.WriteValue(value.ToString());
+                    writer.WriteValue(value == 0 ? "0" : value.ToString());
+                    break;
+                case NumberConversion.Raw:
+                    writer.WriteValue(value);
                     break;
```

Comment:
can you test it in an array? we had issues with serializing arrays when using WriteValue from Json.NET

Reply:
what scenario exactly this is just a primitive?

Reply:
you can leave it -unlikely to happen, Parity trace has an array of ints but it is handled separately

Reply:
unlikely we would end up with an array if nullable longs